src/table.rs

```diff
@@ -1,62 +1,67 @@
+use crate::display_traits::{Show, View};
 use eframe::egui;
 use egui_extras::{Column, Size, StripBuilder, TableBuilder};
 
-pub trait View {
-    fn ui(&mut self, ui: &mut egui::Ui);
-}
-
-/// Something to view
-pub trait PopupTable<T: std::fmt::Display> {
-    /// `&'static` so we can also use it as a key to store open/close state.
-    fn name(&self) -> String;
-    fn set_name(&mut self, name: String);
-    fn set_data(&mut self, data: Vec<T>);
-    /// Show windows, etc
-    fn show(&mut self, ctx: &egui::Context, open: &mut bool);
-}
-
 /// Shows off a table with dynamic layout
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct TableWindow<T> {
     name: String,
-    data: Vec<T>,
+    data: Option<Vec<T>>,
+    scalar: Option<String>,
 }
 
-impl<T> Default for TableWindow<T> {
+impl<T: hdf5::H5Type + std::fmt::Display> Default for TableWindow<T> {
     fn default() -> Self {
         Self {
             name: "Table".to_string(),
-            data: vec![],
+            data: None,
+            scalar: None,
         }
     }
 }
 
-impl<T: std::fmt::Display> PopupTable<T> for TableWindow<T> {
-    fn name(&self) -> String {
-        self.name.clone()
+impl<T: hdf5::H5Type + std::fmt::Display> TableWindow<T> {
+    pub fn set_name(&mut self, name: String) {
+        self.name = name;
     }
 
-    fn set_name(&mut self, name: String) {
-        self.name = name;
+    pub fn set_data(&mut self, data: Vec<T>) {
+        self.data = Some(data);
     }
 
-    fn set_data(&mut self, data: Vec<T>) {
-        self.data = data;
+    pub fn set_scalar(&mut self, scalar: String) {
+        self.scalar = Some(scalar);
     }
+}
 
+impl<T: hdf5::H5Type + std::fmt::Display> Show for TableWindow<T> {
     fn show(&mut self, ctx: &egui::Context, open: &mut bool) {
-        egui::Window::new(self.name())
-            .open(open)
-            .resizable(true)
-            .default_width(100.0)
-            .show(ctx, |ui| {
-                use View as _;
-                self.ui(ui);
-            });
+        if self.data.is_some() {
+            egui::Window::new(&self.name)
+                .open(open)
+                .resizable(true)
+                .default_width(100.0)
+                .show(ctx, |ui| {
+                    use View as _;
+                    self.ui(ui);
+                });
+        } else if self.scalar.is_some() {
+            egui::Window::new(&self.name)
+                .open(open)
+                .vscroll(true)
+                .resizable(true)
+                .default_height(300.0)
+                .show(ctx, |ui| {
+                    ui.label(format!("{:?}", self.scalar.as_ref().unwrap()));
+                });
+        } else {
+            // Nothing to show
+            // Should raise something here?
+        }
     }
 }
 
-impl<T: std::fmt::Display> View for TableWindow<T> {
+impl<T: hdf5::H5Type + std::fmt::Display> View for TableWindow<T> {
```

Comment:
Is the `hdf5::H5Type + ` needed? It seems to be working without it and also without `hdf5::H5Type + ` in table_ui`.

Reply:
I tried to remove it but it fails for me because `ds.read_scalar` and `ds.read_raw` need this trait. But maybe I missed something. If you can make it work don't hesitate to add a new commit :)

Reply:
Ok, just added.